src/result_set.rs

```diff
@@ -484,10 +569,206 @@ impl FromRedisValueWithGraph for Relation {
                         properties,
                     })
                 } else {
-                    server_type_error!("expected array of size 5 as relationship representation",)
+                    server_type_error!("expected array of size 5 as edge representation",)
+                }
+            }
+            _ => server_type_error!("expected array as edge representation"),
+        }
+    }
+}
+
+/// A raw path structure returned by RedisGraph.
+#[derive(Debug, Clone, PartialEq)]
+pub struct RawPath {
+    /// Nodes in the path.
+    pub nodes: Vec<Node>,
+    /// Edges in the path.
+    pub edges: Vec<Edge>,
+}
+
+#[allow(clippy::len_without_is_empty)]
+impl RawPath {
+    /// The length of the path. This is effectively the amount of [`Edge`]s in
+    /// the path.
+    pub fn len(&self) -> usize {
+        self.edges.len()
+    }
+}
+
+impl From<Path> for RawPath {
+    fn from(path: Path) -> Self {
+        let mut nodes: Vec<Node> = Vec::new();
+        let mut edges: Vec<Edge> = Vec::new();
+
+        path.bifor_owned(|node| nodes.push(node), |edge| edges.push(edge));
```

Comment:
For example, this could be `bimap` or such and return `(Vec<Node>, Vec<Edge>)`.

Reply:
Can you elaborate on that? `Path` is not a [`Bifunctor`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Data-Bifunctor.html) since `Node` and `Edge` are fixed. You could easily make `Path` polymorphic and implement `bimap`, but I'm not sure if it would be that useful.

Reply:
I meant more that this code-snippet could be reduced to the following, were `bimap` a method returning `(Vec<A>, Vec<B>)`:

```rust
let (nodes, edges) = path.bimap(|node| node, |edge| edge);
```

Or if there were an alternative to `Iterator` where its methods could handle 2 chains simultaneously, you could do

```rust
let (nodes: Vec<Node>, edges: Vec<Edges>) = path.iter().map(|node| node, |edge| edge).collect();
```

There's nothing actionable here, just thinking from a user-perspective.